libcudacxx/include/cuda/__barrier/barrier_block_scope.h

```diff
@@ -379,12 +379,28 @@ class barrier<thread_scope_block, ::cuda::std::__empty_completion> : public __bl
 public:
   _CCCL_API inline void wait(arrival_token&& __phase) const
   {
+    // no need to back off on a barrier in SMEM on SM90+, SYNCS unit is taking care of this
+    NV_IF_TARGET(NV_PROVIDES_SM_90,
+                 (if (::cuda::device::is_object_from(__barrier, ::cuda::device::address_space::shared)) {
+                   while (!__try_wait(__phase))
```

Comment:
This seems like a common name, should we use
```suggestion
                   while (!::__try_wait(__phase))
```

Reply:
I am calling another member function, so it would be:
```suggestion
                   while (!this->__try_wait(__phase))
```
Do you want to have that change? We are not using `this->` to call member functions in other places.